diagnostic_updater/include/diagnostic_updater/publisher.hpp

```diff
@@ -207,10 +209,10 @@ class DiagnosedPublisher : public TopicDiagnostic
    * The timestamp to be used by the TimeStampStatus class will be
    * extracted from message.header.stamp.
    */
-  virtual void publish(const std::shared_ptr<T> & message)
+  virtual void publish(typename PublisherT::MessageUniquePtr message)
   {
     tick(message->header.stamp);
```

Comment:
I believe this might lead to segfaults if compiling with message types not containing a header

Reply:
I think it just won't compile, but with the assert we get a better error message.

Reply:
You could make it work with header and non-header messages using the `HasHeader` trait, but I'm not sure it's worth the bother:

http://wiki.ros.org/roscpp/Overview/MessagesTraits#Optional_Traits